solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -1618,11 +1620,54 @@ impl pallet_data_preservers::AssignmentPayment<AccountId> for PreserversAssignme
     }
 }
 
+pub struct DataPreserversBenchmarkHelper<T>(PhantomData<T>);
+
+#[cfg(feature = "runtime-benchmarks")]
+impl<T> ArgumentFactory<T::AccountId> for DataPreserversBenchmarkHelper<T>
+where
+    T: pallet_registrar::Config + pallet_balances::Config + paras_registrar::Config,
+{
+    fn reserve_para_id(para_id: ParaId) {
+        use frame_support::{assert_ok, dispatch::RawOrigin, traits::fungible::Mutate};
+
+        let account: T::AccountId = frame_benchmarking::account("para_manager", 0, 1);
+        assert_ok!(<T as pallet_registrar::Config>::Currency::mint_into(
+            &account,
+            <T as pallet_registrar::Config>::Currency::minimum_balance() * 10_000_000u32.into()
+                + T::DepositAmount::get(),
+        ));
+
+        paras_registrar::NextFreeParaId::<T>::put(para_id);
+        assert_eq!(paras_registrar::NextFreeParaId::<T>::get(), para_id);
+        assert_ok!(paras_registrar::Pallet::<T>::reserve(
+            RawOrigin::Signed(account.clone()).into()
+        ));
+
+        let mut storage = vec![];
+        storage.push((b":code".to_vec(), vec![1; 10]).into());
+        let genesis_data = ContainerChainGenesisData {
+            storage,
+            name: Default::default(),
+            id: Default::default(),
+            fork_id: Default::default(),
+            extensions: Default::default(),
+            properties: Default::default(),
+        };
+
+        assert_ok!(pallet_registrar::Pallet::<T>::register(
```

Comment:
do we need to both reserve and register here?

Reply:
Moved reserving ParaId to `InnerDancelightRegister::prepare_chain_registration ` to avoid duplicating code in https://github.com/moondance-labs/tanssi/pull/783/commits/674169290cfa373cac5047de830553dc0e85d3d3